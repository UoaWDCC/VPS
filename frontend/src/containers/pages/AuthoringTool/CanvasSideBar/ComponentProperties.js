import React, { useContext } from "react";
import AuthoringToolContext from "../../../../context/AuthoringToolContext";
import SceneContext from "../../../../context/SceneContext";
import componentPropertiesResolver from "./componentPropertiesResolver";

import styles from "../../../../styling/CanvasSideBar.module.scss";

export default function ComponentProperties() {
  const { select } = useContext(AuthoringToolContext);
  const { currentScene } = useContext(SceneContext);

  console.log(`id: ${select}`);
  console.log(currentScene.components);

  return (
    <>
      <div className={styles.componentPropertiesContainer}>
        <h1 className={styles.sideBarHeader}>Component Properties</h1>
        <div
          className={`${styles.sideBarBody} ${
            select === null ? styles.empty : ""
          }`}
        >
          {select === null ? (
            <p className={styles.noComponentSelectedText}>
              No component selected
            </p>
          ) : (
            componentPropertiesResolver(currentScene.components[select], select)
          )}
        </div>
      </div>
    </>
  );
}
