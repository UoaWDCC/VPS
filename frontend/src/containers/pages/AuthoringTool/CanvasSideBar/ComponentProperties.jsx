import { useContext } from "react";
import AuthoringToolContext from "../../../../context/AuthoringToolContext";
import SceneContext from "../../../../context/SceneContext";
import componentPropertiesResolver from "./componentPropertiesResolver";

import styles from "../../../../styling/CanvasSideBar.module.scss";

/**
 * This component displays the properties the selected scene component
 * @component
 */
export default function ComponentProperties() {
  const { select } = useContext(AuthoringToolContext);
  const { currentScene } = useContext(SceneContext);

  return (
    <>
      <div className={styles.componentPropertiesContainer}>
        <h1 className={styles.sideBarHeader}>Component Properties</h1>
        <div
          className={`${styles.sideBarBody} ${
            select === null ? styles.empty : ""
          }`}
        >
          {select === null ? (
            <p className={styles.noComponentSelectedText}>
              No component selected
            </p>
          ) : (
            componentPropertiesResolver(currentScene.components[select], select)
          )}
        </div>
      </div>
    </>
  );
}
