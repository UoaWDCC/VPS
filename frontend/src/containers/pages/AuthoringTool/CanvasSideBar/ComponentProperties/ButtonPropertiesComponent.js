import React, { useContext } from "react";
import {
  TextField,
  FormControl,
  InputLabel,
  Select,
  MenuItem,
} from "@material-ui/core";
import SceneContext from "../../../../../context/SceneContext";
import CustomTextFieldStyles from "../CustomPropertyInputStyles/CustomTextFieldStyles";
import CustomInputLabelStyles from "../CustomPropertyInputStyles/CustomInputLabelStyles";

import styles from "../../../../../styling/CanvasSideBar.module.scss";

const CustomTextField = CustomTextFieldStyles()(TextField);
const CustomInputLabel = CustomInputLabelStyles()(InputLabel);

export default function ButtonPropertiesComponent({
  component,
  componentIndex,
}) {
  const { scenes, currentScene, setCurrentScene } = useContext(SceneContext);

  function updateComponentProperty(event, property) {
    const updatedComponents = currentScene.components;
    console.log(event.target.value);
    updatedComponents[componentIndex][property] = event.target.value;

    setCurrentScene({
      ...currentScene,
      components: updatedComponents,
    });
  }

  return (
    <>
      <CustomTextField
        label="Text"
        value={component.text}
        fullWidth
        onChange={(event) => updateComponentProperty(event, "text")}
        className={styles.componentProperty}
      />
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Variant</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.variant}
          onChange={(event) => updateComponentProperty(event, "variant")}
        >
          <MenuItem value="contained">Contained</MenuItem>
          <MenuItem value="outlined">Outlined</MenuItem>
        </Select>
      </FormControl>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Colour</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.colour}
          onChange={(event) => updateComponentProperty(event, "colour")}
        >
          <MenuItem value="white">White</MenuItem>
          <MenuItem value="teal">Teal</MenuItem>
        </Select>
      </FormControl>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Next Scene</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.nextScene}
          onChange={(event) => updateComponentProperty(event, "nextScene")}
          displayEmpty
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          {scenes.map((scene) => {
            return (
              <MenuItem key={scene._id} value={scene._id}>
                {scene.name}
              </MenuItem>
            );
          })}
        </Select>
      </FormControl>
    </>
  );
}
