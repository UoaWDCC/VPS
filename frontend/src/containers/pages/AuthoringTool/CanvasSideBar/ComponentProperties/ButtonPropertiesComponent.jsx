import {
  Button,
  Checkbox,
  FormControl,
  FormControlLabel,
  InputLabel,
  MenuItem,
  Select,
  Snackbar,
  TextField,
  Typography,
} from "@material-ui/core";
import { useContext, useState } from "react";
import SceneContext from "../../../../../context/SceneContext";
import CustomInputLabelStyles from "../CustomPropertyInputStyles/CustomInputLabelStyles";
import CustomTextFieldStyles from "../CustomPropertyInputStyles/CustomTextFieldStyles";
import CustomCheckBoxStyles from "../CustomPropertyInputStyles/CustomCheckBoxStyles";

import styles from "../../../../../styling/CanvasSideBar.module.scss";

const CustomTextField = CustomTextFieldStyles()(TextField);
const CustomInputLabel = CustomInputLabelStyles()(InputLabel);
const CustomCheckBox = CustomCheckBoxStyles()(Checkbox);

/**
 * This component displays the properties in the sidebar for a button scene component.
 * @component
 */
export default function ButtonPropertiesComponent({
  component,
  componentIndex,
}) {
  const { scenes, updateComponentProperty } = useContext(SceneContext);

  const [newFlag, setNewFlag] = useState("");
  const [open, setOpen] = useState(false);

  const handleInputChange = (event) => {
    setNewFlag(event.target.value);
  };

  const handleAddFlag = () => {
    if (!component.flagAdditions) {
      updateComponentProperty(componentIndex, "flagAdditions", {});
    }
    if (!component.flagDeletions) {
      updateComponentProperty(componentIndex, "flagDeletions", {});
    }

    if (Object.keys(component.flagAdditions).includes(`${newFlag}`)) {
      setOpen(true);
      return;
    }

    if (newFlag.trim().length !== 0) {
      setNewFlag("");
      updateComponentProperty(componentIndex, "flagAdditions", {
        ...component.flagAdditions,
        [newFlag]: false,
      });
      updateComponentProperty(componentIndex, "flagDeletions", {
        ...component.flagDeletions,
        [newFlag]: false,
      });
    }
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleCheckboxChange = (flag, adding) => {
    if (adding) {
      updateComponentProperty(componentIndex, "flagAdditions", {
        ...component.flagAdditions,
        [flag]: !component.flagAdditions[flag],
      });
    } else {
      updateComponentProperty(componentIndex, "flagDeletions", {
        ...component.flagDeletions,
        [flag]: !component.flagDeletions[flag],
      });
    }
  };

  return (
    <>
      <CustomTextField
        label="Text"
        value={component.text}
        fullWidth
        onChange={(event) =>
          updateComponentProperty(componentIndex, "text", event.target.value)
        }
        className={styles.componentProperty}
      />
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Variant</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.variant}
          onChange={(event) =>
            updateComponentProperty(
              componentIndex,
              "variant",
              event.target.value
            )
          }
        >
          <MenuItem value="contained">Contained</MenuItem>
          <MenuItem value="outlined">Outlined</MenuItem>
        </Select>
      </FormControl>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Colour</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.colour}
          onChange={(event) =>
            updateComponentProperty(
              componentIndex,
              "colour",
              event.target.value
            )
          }
        >
          <MenuItem value="white">White</MenuItem>
          <MenuItem value="teal">Teal</MenuItem>
        </Select>
      </FormControl>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Linked Scene</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.nextScene}
          onChange={(event) =>
            updateComponentProperty(
              componentIndex,
              "nextScene",
              event.target.value
            )
          }
          displayEmpty
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          {scenes.map((scene) => {
            return (
              <MenuItem key={scene._id} value={scene._id}>
                {scene.name}
              </MenuItem>
            );
          })}
        </Select>
      </FormControl>

      <FormControl fullWidth className={styles.componentProperty}>
        <div style={{ display: "flex" }}>
          <CustomTextField
            label="New Flag"
            value={newFlag}
            fullWidth
            onChange={handleInputChange}
          />
          <Button onClick={handleAddFlag}>Add</Button>
        </div>
      </FormControl>

      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink className={styles.plusLabel}>
          Select flags to be added
        </CustomInputLabel>
        <Select
          multiple
          className={styles.selectInput}
          value={Object.keys(component.flagAdditions || {})}
          renderValue={(selected) =>
            selected.filter((flag) => component.flagAdditions[flag]).join(", ")
          }
          displayEmpty
        >
          {Object.keys(component.flagAdditions || {}).length > 0 ? (
            <div style={{ display: "flex", flexDirection: "column" }}>
              {Object.keys(component.flagAdditions).map((flag) => (
                <FormControlLabel
                  key={flag}
                  control={
                    <CustomCheckBox
                      checked={component.flagAdditions[flag]}
                      onChange={() => handleCheckboxChange(flag, true)}
                    />
                  }
                  label={<span className={styles.plusLabel}>{flag}</span>}
                />
              ))}
            </div>
          ) : (
            <Typography className={styles.menuItem}>
              This button property currently does not have any flags. Please add
              flags first using the text field above.
            </Typography>
          )}
        </Select>
      </FormControl>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink className={styles.minusLabel}>
          Select flags to be removed
        </CustomInputLabel>
        <Select
          multiple
          className={styles.selectInput}
          value={Object.keys(component.flagDeletions || {})}
          renderValue={(selected) =>
            selected.filter((flag) => component.flagDeletions[flag]).join(", ")
          }
          displayEmpty
        >
          {Object.keys(component.flagDeletions || {}).length > 0 ? (
            <div style={{ display: "flex", flexDirection: "column" }}>
              {Object.keys(component.flagDeletions).map((flag) => (
                <FormControlLabel
                  key={flag}
                  control={
                    <CustomCheckBox
                      checked={component.flagDeletions[flag]}
                      onChange={() => handleCheckboxChange(flag, false)}
                    />
                  }
                  label={<span className={styles.minusLabel}>{flag}</span>}
                />
              ))}
            </div>
          ) : (
            <Typography className={styles.menuItem}>
              This button property currently does not have any flags. Please add
              flags first using the text field above.
            </Typography>
          )}
        </Select>
        <Snackbar
          open={open}
          autoHideDuration={3000}
          onClose={handleClose}
          message="Flag already exists"
        />
      </FormControl>

      <FormControl fullWidth className={styles.componentProperty}>
        <CustomTextField
          label="Z Axis Position"
          type="number"
          value={component?.zPosition || ""}
          fullWidth
          onChange={(event) =>
            updateComponentProperty(
              componentIndex,
              "zPosition",
              event.target.value
            )
          }
          InputLabelProps={{
            // label moves up whenever there is input
            shrink: !!component.zPosition,
          }}
        />
      </FormControl>
    </>
  );
}
