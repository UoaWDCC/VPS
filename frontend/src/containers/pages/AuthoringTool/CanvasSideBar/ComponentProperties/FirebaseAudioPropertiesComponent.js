import React, { useContext, useEffect, useState } from "react";
import {
  FormControl,
  FormControlLabel,
  Checkbox,
  Button,
  TextField,
} from "@material-ui/core";
import PlayArrowIcon from "@material-ui/icons/PlayArrowRounded";
import PauseIcon from "@material-ui/icons/PauseRounded";
import SceneContext from "../../../../../context/SceneContext";
import CustomCheckBoxStyles from "../CustomPropertyInputStyles/CustomCheckBoxStyles";

import styles from "../../../../../styling/CanvasSideBar.module.scss";
import useStyles from "./FirebaseAudioPropertiesComponent.styles";

const CustomCheckBox = CustomCheckBoxStyles()(Checkbox);

export default function FirebaseAudioPropertiesComponent({
  component,
  componentIndex,
}) {
  const audioComponentStyles = useStyles();

  const { updateComponentProperty } = useContext(SceneContext);
  const [audio] = useState(new Audio(component.url));
  const [playing, setPlaying] = useState(false);

  const toggle = () => setPlaying(!playing);

  useEffect(() => {
    if (playing) {
      audio.play();
    } else {
      audio.pause();
    }
  }, [playing]);

  useEffect(() => {
    audio.addEventListener("ended", () => setPlaying(false));
    return () => {
      audio.removeEventListener("ended", () => setPlaying(false));
    };
  }, []);

  function handleDeselect(e) {
    // Checks if currently focussed on div children
    if (!e.currentTarget.contains(e.relatedTarget)) {
      setPlaying(false);
    }
  }

  return (
    <div onBlur={(e) => handleDeselect(e)}>
      <div className={`${styles.componentProperty}`}>
        {/* <CustomInputLabel shrink>Preview audio</CustomInputLabel> */}
        <TextField
          label="Audio"
          value={component.name}
          fullWidth
          InputProps={{
            readOnly: true,
          }}
        />
        <Button
          className={audioComponentStyles.playButton}
          onClick={toggle}
          size="small"
          color="primary"
          variant="contained"
        >
          {playing ? <PauseIcon /> : <PlayArrowIcon />}
        </Button>
      </div>

      <FormControl fullWidth className={styles.componentProperty}>
        <FormControlLabel
          control={
            <CustomCheckBox
              checked={component.loop}
              color="default"
              onChange={(event) =>
                updateComponentProperty(
                  componentIndex,
                  "loop",
                  event.target.checked
                )
              }
            />
          }
          label="Loop"
        />
      </FormControl>
    </div>
  );
}
