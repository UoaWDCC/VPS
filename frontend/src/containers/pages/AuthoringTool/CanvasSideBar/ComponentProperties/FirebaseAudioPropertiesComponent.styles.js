import { makeStyles } from "@material-ui/core";

export default function useStyles() {
  return makeStyles({
    playButton: {
      float: "right",
      minWidth: "20px",
      gap: "2em",
      marginTop: "1em",
    },
  })();
}
