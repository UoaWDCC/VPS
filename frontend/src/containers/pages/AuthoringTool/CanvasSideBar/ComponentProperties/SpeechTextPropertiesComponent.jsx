import {
  FormControl,
  InputLabel,
  MenuItem,
  OutlinedInput,
  Select,
  TextField,
} from "@material-ui/core";
import FormatAlignCenterIcon from "@material-ui/icons/FormatAlignCenter";
import FormatAlignJustifyIcon from "@material-ui/icons/FormatAlignJustify";
import FormatAlignLeftIcon from "@material-ui/icons/FormatAlignLeft";
import FormatAlignRightIcon from "@material-ui/icons/FormatAlignRight";
import { ToggleButton, ToggleButtonGroup } from "@mui/material";
import AuthoringToolContext from "context/AuthoringToolContext";
import SceneContext from "context/SceneContext";
import { useContext, useEffect, useRef } from "react";

import styles from "styling/CanvasSideBar.module.scss";
import CustomTextFieldStyles from "../CustomPropertyInputStyles/CustomTextFieldStyles";
import CustomInputLabelStyles from "../CustomPropertyInputStyles/CustomInputLabelStyles";
import useStyles from "./TextPropertiesComponent.styles";

const CustomTextField = CustomTextFieldStyles()(TextField);
const CustomInputLabel = CustomInputLabelStyles()(InputLabel);
const sizes = [6, 8, 9, 10, 11, 12, 14, 16, 18, 20, 24, 30, 36, 42, 48, 60, 72];
// export const reference = useRef(null);

/**
 * This component displays the properties in the sidebar for a text scene component.
 * @component
 */
export default function SpeechTextPropertiesComponent({
  component,
  componentIndex,
}) {
  const textComponentStyles = useStyles();
  const { updateComponentProperty } = useContext(SceneContext);

  const { addPropertyRef } = useContext(AuthoringToolContext);
  const textRef = useRef(null);

  useEffect(() => {
    addPropertyRef("text", textRef);
  }, []);

  return (
    <>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Text</CustomInputLabel>
        <OutlinedInput
          className={textComponentStyles.textArea}
          value={component.text}
          inputRef={textRef}
          fullWidth
          multiline
          rows={5}
          onChange={(event) =>
            updateComponentProperty(componentIndex, "text", event.target.value)
          }
        />
      </FormControl>
      <div
        className={`${styles.componentProperty} ${textComponentStyles.inlineRow}`}
      >
        <FormControl>
          <CustomInputLabel shrink>Font size</CustomInputLabel>
          <Select
            className={styles.selectInput}
            value={component.fontSize}
            onChange={(event) =>
              updateComponentProperty(
                componentIndex,
                "fontSize",
                event.target.value
              )
            }
          >
            {sizes.map((size) => {
              return (
                <MenuItem value={size} key={size}>
                  {size}
                </MenuItem>
              );
            })}
          </Select>
        </FormControl>
        <ToggleButtonGroup
          value={component.textAlign}
          exclusive
          size="small"
          onChange={(event, value) =>
            updateComponentProperty(componentIndex, "textAlign", value)
          }
          aria-label="text alignment"
        >
          <ToggleButton value="left" aria-label="left aligned">
            <FormatAlignLeftIcon />
          </ToggleButton>
          <ToggleButton value="center" aria-label="centered">
            <FormatAlignCenterIcon />
          </ToggleButton>
          <ToggleButton value="right" aria-label="right aligned">
            <FormatAlignRightIcon />
          </ToggleButton>
          <ToggleButton value="justify" aria-label="justified">
            <FormatAlignJustifyIcon />
          </ToggleButton>
        </ToggleButtonGroup>
      </div>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Text Colour</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.color}
          onChange={(event) =>
            updateComponentProperty(componentIndex, "color", event.target.value)
          }
        >
          <MenuItem value="black">Black</MenuItem>
          <MenuItem value="green">Green</MenuItem>
          <MenuItem value="red">Red</MenuItem>
          <MenuItem value="white">White</MenuItem>
        </Select>
      </FormControl>

      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Arrow Location</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.arrowLocation}
          onChange={(event) =>
            updateComponentProperty(
              componentIndex,
              "arrowLocation",
              event.target.value
            )
          }
        >
          <MenuItem value="top">Top</MenuItem>
          <MenuItem value="left">Left</MenuItem>
          <MenuItem value="right">Right</MenuItem>
          <MenuItem value="bottom">Bottom</MenuItem>
        </Select>
      </FormControl>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomTextField
          label="Z Axis Position"
          type="number"
          value={component?.zPosition || ""}
          fullWidth
          onChange={(event) =>
            updateComponentProperty(
              componentIndex,
              "zPosition",
              event.target.value
            )
          }
          InputLabelProps={{
            // label moves up whenever there is input
            shrink: !!component.zPosition,
          }}
        />
      </FormControl>
    </>
  );
}
