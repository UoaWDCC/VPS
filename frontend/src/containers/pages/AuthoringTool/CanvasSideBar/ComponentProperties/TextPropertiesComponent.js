import React, { useContext } from "react";
import {
  TextField,
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  FormControlLabel,
  Checkbox,
} from "@material-ui/core";
import { ToggleButton, ToggleButtonGroup } from "@mui/material";
import FormatAlignLeftIcon from "@material-ui/icons/FormatAlignLeft";
import FormatAlignCenterIcon from "@material-ui/icons/FormatAlignCenter";
import FormatAlignRightIcon from "@material-ui/icons/FormatAlignRight";
import FormatAlignJustifyIcon from "@material-ui/icons/FormatAlignJustify";
import SceneContext from "../../../../../context/SceneContext";
import CustomTextFieldStyles from "../CustomPropertyInputStyles/CustomTextFieldStyles";
import CustomInputLabelStyles from "../CustomPropertyInputStyles/CustomInputLabelStyles";
import CustomCheckBoxStyles from "../CustomPropertyInputStyles/CustomCheckBoxStyles";

import styles from "../../../../../styling/CanvasSideBar.module.scss";
import useStyles from "./TextPropertiesComponent.styles";

const CustomTextField = CustomTextFieldStyles()(TextField);
const CustomInputLabel = CustomInputLabelStyles()(InputLabel);
const CustomCheckBox = CustomCheckBoxStyles()(Checkbox);
const sizes = [6, 8, 9, 10, 11, 12, 14, 16, 18, 20, 24, 30, 36, 42, 48, 60, 72];

export default function TextPropertiesComponent({ component, componentIndex }) {
  const textComponentStyles = useStyles();
  const { currentScene, setCurrentScene } = useContext(SceneContext);

  function updateComponentProperty(property, newValue) {
    const updatedComponents = currentScene.components;
    updatedComponents[componentIndex][property] = newValue;

    setCurrentScene({
      ...currentScene,
      components: updatedComponents,
    });
  }

  // Potential properties
  // Text
  // Border
  // font size
  // font color
  // alignment
  // bold/italics?

  return (
    <>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Text</CustomInputLabel>
        <CustomTextField
          className={textComponentStyles.textArea}
          value={component.text}
          fullWidth
          multiline
          variant="outlined"
          rows={5}
          onChange={(event) =>
            updateComponentProperty("text", event.target.value)
          }
        />
      </FormControl>
      <div
        className={`${styles.componentProperty} ${textComponentStyles.inlineRow}`}
      >
        <FormControl>
          <CustomInputLabel shrink>Font size</CustomInputLabel>
          <Select
            className={styles.selectInput}
            value={component.fontSize}
            onChange={(event) =>
              updateComponentProperty("fontSize", event.target.value)
            }
          >
            {sizes.map((size) => {
              return <MenuItem value={size}>{size}</MenuItem>;
            })}
          </Select>
        </FormControl>
        <ToggleButtonGroup
          value={component.textAlign}
          exclusive
          size="small"
          onChange={(event, value) =>
            updateComponentProperty("textAlign", value)
          }
          aria-label="text alignment"
        >
          <ToggleButton value="left" aria-label="left aligned">
            <FormatAlignLeftIcon />
          </ToggleButton>
          <ToggleButton value="center" aria-label="centered">
            <FormatAlignCenterIcon />
          </ToggleButton>
          <ToggleButton value="right" aria-label="right aligned">
            <FormatAlignRightIcon />
          </ToggleButton>
          <ToggleButton value="justify" aria-label="justified">
            <FormatAlignJustifyIcon />
          </ToggleButton>
        </ToggleButtonGroup>
      </div>
      <FormControl fullWidth className={styles.componentProperty}>
        <CustomInputLabel shrink>Text Colour</CustomInputLabel>
        <Select
          className={styles.selectInput}
          value={component.color}
          onChange={(event) =>
            updateComponentProperty("color", event.target.value)
          }
        >
          <MenuItem value="black">Black</MenuItem>
          <MenuItem value="green">Green</MenuItem>
          <MenuItem value="red">Red</MenuItem>
          <MenuItem value="white">White</MenuItem>
        </Select>
      </FormControl>
      <FormControl fullWidth className={styles.componentProperty}>
        <FormControlLabel
          control={
            <CustomCheckBox
              checked={component.border}
              color="default"
              onChange={(event) =>
                updateComponentProperty("border", event.target.checked)
              }
            />
          }
          label="Include border"
        />
      </FormControl>
    </>
  );
}
