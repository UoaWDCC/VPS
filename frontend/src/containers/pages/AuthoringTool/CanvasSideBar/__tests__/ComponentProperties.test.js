import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import AuthenticationContextProvider from "../../../../../context/AuthenticationContextProvider";
import AuthoringToolContextProvider from "../../../../../context/AuthoringToolContextProvider";
import ScenarioContextProvider from "../../../../../context/ScenarioContextProvider";
import SceneContextProvider from "../../../../../context/SceneContextProvider";
import ComponentProperties from "../ComponentProperties";

test("Scenario Selection page snapshot test", () => {
  const component = renderer.create(
    <BrowserRouter>
      <AuthenticationContextProvider>
        <ScenarioContextProvider>
          <SceneContextProvider>
            <AuthoringToolContextProvider>
              <ComponentProperties />
            </AuthoringToolContextProvider>
          </SceneContextProvider>
        </ScenarioContextProvider>
      </AuthenticationContextProvider>
    </BrowserRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
