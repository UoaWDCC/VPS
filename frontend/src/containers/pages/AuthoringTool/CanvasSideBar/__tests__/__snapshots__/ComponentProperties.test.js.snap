// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenario Selection page snapshot test 1`] = `
<div
  className="componentPropertiesContainer"
>
  <h1
    className="sideBarHeader"
  >
    Properties
  </h1>
  <div
    className="sideBarBody empty"
  >
    <p
      className="noComponentSelectedText"
    >
      No component selected
    </p>
  </div>
</div>
`;
